Build Failure on refs/heads/main

@gemini-code-assist /jules debug this build error

**Log:**
```
ILED
> Task :opencv:bundleDebugAar
> Task :core:nativebridge:buildCMakeDebug[arm64-v8a]
> Task :core:common:compileDebugKotlin
gradle/actions: Writing build results to /home/runner/work/_temp/.gradle-actions/build-results/gradle-build-1776783028508.json

[Incubating] Problems report is available at: file:///home/runner/work/GraffitiXR/GraffitiXR/build/reports/problems/problems-report.html

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':android_collaboration_module:buildCMakeDebug[arm64-v8a]'.
> com.android.ide.common.process.ProcessException: ninja: Entering directory `/home/runner/work/GraffitiXR/GraffitiXR/collab/.cxx/Debug/6542241j/arm64-v8a'
  [1/2] Building CXX object CMakeFiles/collaboration_bridge.dir/CollaborationBridge.cpp.o
  [2/2] Linking CXX shared library /home/runner/work/GraffitiXR/GraffitiXR/collab/build/intermediates/cxx/Debug/6542241j/obj/arm64-v8a/libcollaboration_bridge.so
  FAILED: /home/runner/work/GraffitiXR/GraffitiXR/collab/build/intermediates/cxx/Debug/6542241j/obj/arm64-v8a/libcollaboration_bridge.so 
  : && /usr/local/lib/android/sdk/ndk/28.2.13676358/toolchains/llvm/prebuilt/linux-x86_64/bin/clang++ --target=aarch64-none-linux-android29 --sysroot=/usr/local/lib/android/sdk/ndk/28.2.13676358/toolchains/llvm/prebuilt/linux-x86_64/sysroot -fPIC -g -DANDROID -fdata-sections -ffunction-sections -funwind-tables -fstack-protector-strong -no-canonical-prefixes -D_FORTIFY_SOURCE=2 -Wformat -Werror=format-security  -std=c++17 -fno-limit-debug-info  -static-libstdc++ -Wl,--build-id=sha1 -Wl,--no-rosegment -Wl,--no-undefined-version -Wl,--fatal-warnings -Wl,--no-undefined -Qunused-arguments -shared -Wl,-soname,libcollaboration_bridge.so -o /home/runner/work/GraffitiXR/GraffitiXR/collab/build/intermediates/cxx/Debug/6542241j/obj/arm64-v8a/libcollaboration_bridge.so CMakeFiles/collaboration_bridge.dir/CollaborationBridge.cpp.o  /usr/local/lib/android/sdk/ndk/28.2.13676358/toolchains/llvm/prebuilt/linux-x86_64/sysroot/usr/lib/aarch64-linux-android/29/liblog.so  -latomic -lm && :
  ld.lld: error: undefined symbol: mobilegs::exportFingerprint()
  >>> referenced by CollaborationBridge.cpp:22 (/home/runner/work/GraffitiXR/GraffitiXR/collab/src/main/cpp/CollaborationBridge.cpp:22)
  >>>               CMakeFiles/collaboration_bridge.dir/CollaborationBridge.cpp.o:(Java_com_hereliesaz_graffitixr_core_collaboration_CollaborationManager_nativeExportFingerprint)
  
  ld.lld: error: undefined symbol: mobilegs::alignToFingerprint(unsigned char const*, unsigned long)
  >>> referenced by CollaborationBridge.cpp:37 (/home/runner/work/GraffitiXR/GraffitiXR/collab/src/main/cpp/CollaborationBridge.cpp:37)
  >>>               CMakeFiles/collaboration_bridge.dir/CollaborationBridge.cpp.o:(Java_com_hereliesaz_graffitixr_core_collaboration_CollaborationManager_nativeAlignToPeer)
  clang++: error: linker command failed with exit code 1 (use -v to see invocation)
  ninja: build stopped: subcommand failed.
  
  C++ build system [build] failed while executing:
      /usr/local/lib/android/sdk/cmake/3.22.1/bin/ninja \
        -C \
        /home/runner/work/GraffitiXR/GraffitiXR/collab/.cxx/Debug/6542241j/arm64-v8a \
        collaboration_bridge
    from /home/runner/work/GraffitiXR/GraffitiXR/collab

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights from a Build Scan (powered by Develocity).
> Get more help at https://help.gradle.org.

Deprecated Gradle features were used in this build, making it incompatible with Gradle 10.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/9.4.0-rc-1/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

BUILD FAILED in 57s
165 actionable tasks: 76 executed, 89 from cache

```

--- a/core/nativebridge/src/main/cpp/GraffitiJNI.cpp
+++ b/core/nativebridge/src/main/cpp/GraffitiJNI.cpp
@@ -576,4 +576,28 @@ Java_com_hereliesaz_graffitixr_nativebridge_SlamManager_nativeUnrollMesh(JNIEnv*
     return result;
 }
 
+JNIEXPORT jbyteArray JNICALL
+Java_com_hereliesaz_graffitixr_nativebridge_SlamManager_nativeExportFingerprint(
+        JNIEnv* env, jobject thiz) {
+    if (!gSlamEngine) return nullptr;
+    std::vector<uint8_t> fingerprint = gSlamEngine->exportFingerprint();
+    if (fingerprint.empty()) return nullptr;
+
+    jbyteArray result = env->NewByteArray(fingerprint.size());
+    env->SetByteArrayRegion(result, 0, fingerprint.size(), (jbyte*)fingerprint.data());
+    return result;
+}
+
+JNIEXPORT void JNICALL
+Java_com_hereliesaz_graffitixr_nativebridge_SlamManager_nativeAlignToFingerprint(
+        JNIEnv* env, jobject thiz, jbyteArray data) {
+    if (!gSlamEngine) return;
+    jsize size = env->GetArrayLength(data);
+    jbyte* buffer = env->GetByteArrayElements(data, nullptr);
+
+    gSlamEngine->alignToFingerprint((uint8_t*)buffer, size);
+
+    env->ReleaseByteArrayElements(data, buffer, JNI_ABORT);
+}
+
 } // extern "C"

--- a/core/nativebridge/src/main/java/com/hereliesaz/graffitixr/nativebridge/SlamManager.kt
+++ b/core/nativebridge/src/main/java/com/hereliesaz/graffitixr/nativebridge/SlamManager.kt
@@ -161,6 +161,9 @@ class SlamManager @Inject constructor() {
     fun getPersistentMesh(vertices: FloatArray, weights: FloatArray) = nativeGetPersistentMesh(vertices, weights)
     fun unrollMesh(vertices: FloatArray): FloatArray = nativeUnrollMesh(vertices)
 
+    fun exportFingerprint(): ByteArray? = nativeExportFingerprint()
+    fun alignToFingerprint(data: ByteArray) = nativeAlignToFingerprint(data)
+
     fun destroy() {
         if (isInitialized) {
             nativeDestroy()
@@ -210,6 +213,8 @@ class SlamManager @Inject constructor() {
     private external fun nativeSetMuralMethod(method: Int)
     private external fun nativeGetPersistentMesh(vertices: FloatArray, weights: FloatArray)
     private external fun nativeUnrollMesh(vertices: FloatArray): FloatArray
+    private external fun nativeExportFingerprint(): ByteArray?
+    private external fun nativeAlignToFingerprint(data: ByteArray)
     private external fun nativeSetWallFingerprint(
         bitmap: Bitmap, mask: Bitmap?,
         depthBuffer: ByteBuffer,

